kafka-rest/src/test/java/io/confluent/kafkarest/integration/CustomLogIntegrationTest.java

```diff
@@ -0,0 +1,515 @@
+import io.confluent.kafkarest.entities.v3.ProduceRequest;
+import io.confluent.kafkarest.entities.v3.ProduceRequest.ProduceRequestData;
+import io.confluent.kafkarest.ratelimit.RateLimitExceededException.ErrorCodes;
+import io.confluent.kafkarest.requestlog.CustomLogRequestAttributes;
+import java.time.Duration;
+import java.util.List;
+import java.util.Properties;
+import java.util.concurrent.ConcurrentLinkedQueue;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import javax.ws.rs.client.Entity;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import org.eclipse.jetty.server.RequestLog;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInfo;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Integration tests to validate request-log has necessary custom-log when request is rate-limited.
+ */
+@Tag("IntegrationTest")
+public class CustomLogIntegrationTest extends ClusterTestHarness {
+
+  private static final Logger log = LoggerFactory.getLogger(CustomLogIntegrationTest.class);
+
+  private ScheduledExecutorService executor = Executors.newScheduledThreadPool(4);
+
+  private final ConcurrentLinkedQueue<String> logEntries = new ConcurrentLinkedQueue<>();
+
+  private static final String topicName = "topic-1";
+
+  private Properties restConfigs = new Properties();
+
+  public CustomLogIntegrationTest() {
+    super(1, false, false);
+  }
+
+  @Override
+  protected void overrideKafkaRestConfigs(Properties restProperties) {
+    restProperties.putAll(restConfigs);
+  }
+
+  @Override
+  /** Override to do nothing, as this class has an overloaded setup() below. */
+  public void setUp() {}
+
+  @BeforeEach
+  public void setUp(TestInfo testInfo) throws Exception {
+    super.setUp();
+
+    restConfigs.clear();
+    // Adding custom KafkaRestConfigs for individual test-cases/test-methods below.
+    // Make the "other" rate-limit large to make sure they doesn't trigger.
+    restConfigs.put("dos.filter.delay.ms", -1);
+    if (testInfo
+        .getDisplayName()
+        .contains("test_whenJettyGlobalRateLimitTriggered_ThenRequestLogHasRelevantInfo")) {
+      restConfigs.put("dos.filter.enabled", "true");
+      restConfigs.put("dos.filter.max.requests.per.sec", 1);
+      restConfigs.put("dos.filter.max.requests.per.connection.per.sec", 99999);
+    } else if (testInfo
+        .getDisplayName()
+        .contains("test_whenNoRateLimitTriggered_ThenRequestLogHasRelevantInfo")) {
+      // Set all rate-limits very high
+      restConfigs.put("dos.filter.enabled", "true");
```

Comment:
can you use the constant from RestConfig class?

Reply:
Unfortunately constant is private, with no setter 😢

Reply:
😢